docs/compilation.rst

```diff
@@ -378,3 +374,74 @@ To run Python unit tests:
     pip install pytest
     make install-pip-package -j$(nproc)
     pytest ../python/test
+
+.. _ccache:
+
+Caching compilation with ccache
+-------------------------------
+
+ccache is a compiler cache that can speed up the compilation process by avoiding
+recompilation of the same source code. It can significantly speed up
+recompilation of Open3D on Linux/macOS, even if you clear the ``build``
+directory. You'll need ccache 4.0+ to cache both C++ and CUDA compilations.
+
+After installing ``ccache``, simply reconfigure and recompile the Open3D
+library. Open3D's CMake script can detect and use it automatically. You don't
+need to setup additional paths except for the ``ccache`` program itself.
+
+Ubuntu 18.04, 20.04
+```````````````````
+
+If you install ``ccache`` via ``sudo apt install ccache``, the 3.x version will
+be installed. To cache CUDA compilations, you'll need the 4.0+ version. Here, we
+demonstrate one way to setup ``ccache`` by compiling it from source, installing
+it to ``${HOME}/bin``, and adding ``${HOME}/bin`` to ``${PATH}``.
+
+.. code-block:: bash
+
+    git clone https://github.com/ccache/ccache.git
+    cd ccache
+    git checkout v4.6 -b 4.6
+
+    mkdir build
+    cd build
+    cmake -DZSTD_FROM_INTERNET=ON \
+          -DHIREDIS_FROM_INTERNET=ON \
+          -DCMAKE_BUILD_TYPE=Release \
+          -DCMAKE_INSTALL_PREFIX=$HOME \
+          ..
+    make -j$(nproc)
+    make install -j$(nproc)
+
+Now, open your ``~/.bashrc`` file and add the following line:
+
+.. code-block:: bash
+    PATH=${HOME}:${PATH}
```

Comment:
Instead of asking the user to open the bashrc and add the line, can we do:
```bash
echo 'PATH=${HOME}:${PATH}' >> ~/.bashrc 
```
and instead of restarting the terminal, we may do: `bash` or `source ~/.bashrc`. 

In summary, 
```bash
    # Clone the ccache source.
    git clone https://github.com/ccache/ccache.git
    cd ccache
    git checkout v4.6 -b 4.6
    # Build and install the ccache from source.
    mkdir build
    cd build
    cmake -DZSTD_FROM_INTERNET=ON \
          -DHIREDIS_FROM_INTERNET=ON \
          -DCMAKE_BUILD_TYPE=Release \
          -DCMAKE_INSTALL_PREFIX=$HOME \
          ..
    make -j$(nproc)
    make install -j$(nproc)
    
    # Add the path to your bashrc.
    echo 'PATH=${HOME}:${PATH}' >> ~/.bashrc 
    
    # Reload your terminal (or restart the terminal)
    source ~/.bashrc
    
    # Verify ``ccache`` has been installed correctly
    which ccache
    ccache --version   
```
The output must look similar to:
```bash
/usr/local/bin/ccache
    
ccache version 4.4
Features: file-storage http-storage redis-storage

Copyright (C) 2002-2007 Andrew Tridgell
Copyright (C) 2009-2021 Joel Rosdahl and other contributors

See <https://ccache.dev/credits.html> for a complete list of contributors.

This program is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free Software
Foundation; either version 3 of the License, or (at your option) any later
version.
```

Reply:
Done